tests/rustdoc-json/attrs/inline.rs

```diff
@@ -1,11 +1,11 @@
-//@ is "$.index[?(@.name=='just_inline')].attrs" '["#[inline]"]'
+//@ is "$.index[?(@.name=='just_inline')].attrs" '["#[attr = Inline(Hint)]"]'
```

Comment:
So to understand https://github.com/rust-lang/rust/pull/138165/files?diff=unified&w=1#r2143839007 right, we could represent this `attrs` list in json not as a list of strings, but as a list of either strings (unstable format, only print or view it) or json maps that contain all the information in a structured way and doesn't require consumers to re-parse it.

If we change the map representation, we bump the format. If we change the string representation, we do not.

Either way, we either break csc or we bump the version so csc can gate its parser on it. The only question we're asking right now is whether we bump the version twice (once now, and once for when we move to json maps).

is that about right @aDotInTheVoid @obi1kenobi ?

Reply:
From a c-s-c perspective I don't have a strong opinion, we don't check `#[inline]` and the change shouldn't break us.

From a rustdoc JSON format perspective, I think it's kinder to other rustdoc JSON users if we bump.